### DISCUSSION TITLE
Allow targets with format = "file" that sometimes do not create a file

### DISCUSSION THREAD

**Discussion by programLyrique** [ropensci/targets#728] [Help] [answered]
My pipeline has a dynamic target that generates a bunch of files, but possibly none.

In the case when one of the branches generates no files, the pipeline stops with the following error:

```
Error : _store_ a target must have at least one output file.
```

The documentation says it is the intended behaviour indeed for external output files:

> the return value is a character value of paths to existing files and nonempty directories. 

A pain point in that case is it does not seem to be an error in the target but a failing assertion of the targets package itself so it is very complicated to find out which branch did not create any file: `error = "continue"` does not work, as well saving a workspace on error.

I thought of a solution which sounds more like a hack: always creating a dummy file in addition to the possible real ones, and then filter it out of the list of paths in a child target.

Would it be possible to allow targets that might not create any files or just empty directories?

**Comment by wlandau** [comment, marked as answer]
This is actually doable. Implemented in 08cd0542acd376e565d75f1446d54f300a41a86f. Empty file targets can now return `character(0)`.

**Reply by programLyrique** [reply]
Thank you!! (and I am impressed by how quickly you answered!)
